Mathlib/Data/Polynomial/Expand.lean

```diff
@@ -170,9 +170,15 @@ theorem natDegree_expand (p : ℕ) (f : R[X]) : (expand R p f).natDegree = f.nat
     exact mt leadingCoeff_eq_zero.1 hf
 #align polynomial.nat_degree_expand Polynomial.natDegree_expand
 
-theorem Monic.expand {p : ℕ} {f : R[X]} (hp : 0 < p) (h : f.Monic) : (expand R p f).Monic := by
-  rw [Monic.def, Polynomial.leadingCoeff, natDegree_expand, coeff_expand hp]
-  simp [hp, h]
+theorem leadingCoeff_expand {p : ℕ} {f : R[X]} (hp : 0 < p) :
+    (expand R p f).leadingCoeff = f.leadingCoeff := by
+  simp only [leadingCoeff, natDegree_expand, coeff_expand hp, dvd_mul_left, ite_true,
+    Nat.mul_div_cancel _ hp]
+
+theorem monic_expand_iff {p : ℕ} {f : R[X]} (hp : 0 < p) : (expand R p f).Monic ↔ f.Monic := by
```

Comment:
I'm not sure about this, since this makes `rw [monic_expand_iff hp]` becomes `rw [monic_expand_iff _ hp]`, and breaks the following `alias ... = monic_expand_iff` (the original `Monic.expand` has `f` implicit).

Reply:
Yeah I think we it's conventional to make variables rewrite (Eq or Iff) lemmas implicit if they appear in both sides.